app/assets/src/components/views/SampleView/SampleView.jsx

```diff
@@ -66,19 +74,30 @@ class SampleView extends React.Component {
     return "gsnap filter on human/chimp genome was not run.";
   };
 
-  refreshPage = overrideUrlParams => {
+  refreshPage = (overrideUrlParams, reload = true) => {
     const newParams = Object.assign(
       {},
       this.props.reportPageParams,
       overrideUrlParams
     );
-    window.location =
+    const url =
       location.protocol +
       "//" +
       location.host +
       location.pathname +
       "?" +
       getURLParamString(newParams);
+    if (reload) {
```

Comment:
Seems like this snippet can be moved to a utils file too, no? 
You used it in multiple places, right?

Reply:
This is the only place for now. But yes in general it would be good to decide how we want URL history to work everywhere and rely on a single function for it.